examples/minimal_client_service/src/minimal_client_async.rs

```diff
@@ -1,12 +1,10 @@
-use std::env;
-
 use anyhow::{Error, Result};
 
 #[tokio::main]
 async fn main() -> Result<(), Error> {
-    let context = rclrs::Context::new(env::args())?;
+    let mut executor = rclrs::Context::default_from_env()?.create_basic_executor();
```

Comment:
@mxgrey how would a custom executor work here? It'd take a `Context` as a argument, right? I'd remove the `create_basic_executor` function and be slightly more verbose (`SingleThreadedExecutor::new(&context)`) to show how the API works and basically to not make any distinction between the executors.